src/js/actions/export.js

```diff
@@ -0,0 +1,365 @@
+     * @private
+     * @param {Document} document Owner document
+     * @param {Layer|Immutable.List.<Layer>} layers Either a Layer reference or array of Layers
+     * @param {boolean=} exportEnabled
+     * @returns {Promise}
+     */
+    var setLayerExportEnabled = function (document, layers, exportEnabled) {
+        var layerIDs = Immutable.List.isList(layers) ?
+                collection.pluck(layers, "id").toArray() :
+                [layers.id];
+
+        var payload = {
+                documentID: document.id,
+                layerIDs: layerIDs,
+                exportEnabled: exportEnabled
+            };
+
+        this.dispatch(events.document.LAYER_EXPORT_ENABLED_CHANGED, payload);
+
+        return Promise.all(layerIDs.map(function (layerID) {
+            return _syncLayerExportMetadata.call(this, document.id, layerID);
+        }.bind(this)));
+    };
+    setLayerExportEnabled.reads = [locks.JS_DOC];
+    setLayerExportEnabled.writes = [locks.JS_DOC, locks.PS_DOC];
+
+    /**
+     * Helper function to export a single asset using the export service, and update the metadata afterwards
+     *
+     * @private
+     * @param {Document} document
+     * @param {Layer} layer
+     * @param {number} assetIndex index of this asset within the layer's list
+     * @param {ExportAsset} asset asset to export
+     * @return {Promise}
+     */
+    var _exportAsset = function (document, layer, assetIndex, asset) {
+        return _exportService.exportLayerAsset(layer, asset)
+            .bind(this)
+            .then(function (pathArray) {
+                // Do we need to be aware of exports that return >1 file path?
+                var assetProps = {
+                    filePath: pathArray[0],
+                    status: ExportAsset.STATUS.STABLE
+                };
+                return this.transfer(updateLayerExportAsset, document, layer, assetIndex, assetProps);
+            });
+    };
+
+    /**
+     * Export all assets for the given document for which export has been enabled (layer.exportEnabled)
+     *
+     * @param {Document} document
+     * @return {Promise} resolves when all assets have been exported
+     */
+    var exportAllAssets = function (document) {
+        if (!document) {
+            Promise.resolve("No Document");
+        }
+
+        if (!_exportService || !_exportService.ready()) {
+            Promise.reject("Export Service is not available");
+        }
+
+        var documentID = document.id,
+            documentExports = this.flux.stores.export.getDocumentExports(documentID),
+            layerExportsMap = documentExports && documentExports.layerExportsMap,
+            exportArray = [];
+
+        if (!layerExportsMap || layerExportsMap.size < 1) {
+            return Promise.resolve("no assets to export");
+        }
+
+        // Iterate over the exports map, find the associated layer, test of "exportEnabled"
+        layerExportsMap.forEach(function (layerExportAssets, layerID) {
```

Comment:
Seems like the intermediate `layers.byID(layerID)` and subsequent exportEnabled test makes it more efficient and clear as written, but I'm open to schoolin'